### PR TITLE
Issue #36

### PR DESCRIPTION
**Pull request by PedroL33**
Added alphabetical ordering filter.

### CODE REVIEW

File: src/store.ts [hunk 1]
```diff
@@ -20,14 +21,16 @@ export const updateTagsCountMutation = "updateTagsCount";
...
+export const updateOrderFilterMutation = "updateOrderFilter";
 export const updateFilterSearchPhraseMutation = "updateFilterSearchPhrase";
+
 export const updateSelectedCategoriesMutation = "updateSelectedCategories";
 export const updateSelectedKenticoVersionMutation =
   "updateSelectedKenticoVersion";
 export const toggleCategoryInSelectedCategoriesMutation =
   "toggleCategoryInSelectedCategories";
-
+  export const updateSelectedOrderMutation = "updateSelectedOrder";
```
Comment (makma) [inline src/store.ts:32]:
```suggestion
 export const updateSelectedOrderMutation = "updateSelectedOrder";
```

File: src/utils/filter.ts [hunk 2]
```diff
@@ -18,8 +20,12 @@ export default function performItemsFiltering() {
     categoryFilteredItems,
     selectedKenticoVersion
   );
+  const orderedItems = applyOrderFilter(
+    itemsFilteredByKenticoVersion,
+    selectedOrderFilter
+  )
```
Comment (makma) [inline src/utils/filter.ts:26]:
```suggestion
  );
```

File: src/utils/order.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import store, { updateOrderFilterMutation } from "@/store";
+
+export function initStoreWithOrders() {
+    let orderTypes = ["Random sort", "A-Z", "Z-A"]
```
Comment (makma) [inline src/utils/order.ts:4]:
```suggestion
    let orderTypes = ["Random sort", "A - Z", "Z - A"];
```

File: src/utils/order.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+    store.commit(updateOrderFilterMutation, orderTypes)
```
Comment (makma) [inline src/utils/order.ts:5]:
```suggestion
    store.commit(updateOrderFilterMutation, orderTypes);
```

File: src/utils/filter.ts [hunk 5]
```diff
@@ -78,3 +84,20 @@ function doesItemSupportKenticoVersion(
   );
   return foundVersions && foundVersions.length > 0;
 }
+
+function applyOrderFilter(
+  itemsToFilter: Array<MarketplaceItemModel>,
+  selectedOrderFilter: string
+) {
+  var items = itemsToFilter;
+  switch(selectedOrderFilter) {
```
Comment (makma) [inline src/utils/filter.ts:93]:
```suggestion
  switch (selectedOrderFilter) {
```

File: src/utils/filter.ts [hunk 6]
```diff
@@ -78,3 +84,20 @@ function doesItemSupportKenticoVersion(
...
+    case "Random sort":
+      return itemsToFilter;
+    case "A-Z":
+      return items.slice().sort((a,b) => a.name > b.name ? 1 : -1);
```
Comment (makma) [inline src/utils/filter.ts:97]:
```suggestion
      return items.slice().sort((a, b) => a.name > b.name ? 1 : -1);
```

File: src/utils/filter.ts [hunk 7]
```diff
@@ -78,3 +84,20 @@ function doesItemSupportKenticoVersion(
...
+    case "Z-A":
+      return items.slice().sort((a,b) => b.name > a.name ? 1 : -1);
```
Comment (makma) [inline src/utils/filter.ts:99]:
```suggestion
      return items.slice().sort((a, b) => b.name > a.name ? 1 : -1);
```

File: src/components/OrderFilter.vue [hunk 8]
```diff
@@ -0,0 +1,99 @@
+<template>
+  <div class="order-filter">
+    <div class="selected" @click="toggle">
+      <span class="selected--title">{{ selectedOrder }}</span>
+      <span class="dropdown-icon">▼</span>
+    </div>
+    <div class="options" v-show="isOpen">
+      <div
+      class="option"
```
Comment (makma) [inline src/components/OrderFilter.vue:9]: Pls indent HTML attributes here.

File: src/store.ts [hunk 9]
```diff
@@ -36,6 +39,7 @@ export const addPageAction = "addPage";
 export const updateSearchPhraseAction = "updateFilterPassphrase";
 export const updateSelectedCategoriesAction = "updateSelectedCategories";
 
+
```
Comment (makma) [inline src/store.ts:42]: An unnecessary empty line.

File: src/utils/filter.ts [hunk 10]
```diff
@@ -78,3 +84,20 @@ function doesItemSupportKenticoVersion(
   );
   return foundVersions && foundVersions.length > 0;
 }
+
+function applyOrderFilter(
+  itemsToFilter: Array<MarketplaceItemModel>,
+  selectedOrderFilter: string
+) {
+  var items = itemsToFilter;
```
Comment (makma) [inline src/utils/filter.ts:92]:
```suggestion
  const items = itemsToFilter;
```

**Comment by makma** [comment]
Hi @PedroL33  thank you for your PR! It looks and works really well.

I've added some comments and suggestions with minor changes. Please take a look at them.

I've got, from my point of view, one important suggestion. From my experience, it's always tricky with `order` naming for two reasons:

1. Some programming languages have `order` as a reserved word,
1. `Order` might be confusing and ambiguous if the site contains e-commece features and orders.

I know this is not our case (order is not reserved word in TS, HTML, or CSS) and our site does not have e-commerce orders yet, however, I'd recommend changing `order` naming in code to e.g. `ordering`.

What do you think?

**Comment by PedroL33** [comment]
Thanks for the suggestions.  I just pushed some new commits with the changes you had in mind.

**Pull request merged**